### PR TITLE
Auto-create pgsm for new dbs

### PR DESCRIPTION
**Pull request by nastena1606**
new file:   docs/auto-enable.md
	modified:   docs/setup.md
	modified:   mkdocs.yml

### CODE REVIEW

File: docs/auto-enable.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+1. Create a database to serve as the template.
+
+    ```{.bash data-prompt="$"}
+    $ createdb my-template
+    ```
+
+2. Connect to this database and create the `pg_stat_monitor` extension
+
+    ```{.bash data-prompt="$"}
+    $ psql -d my-template -c 'create extension pg_stat_monitor;'
```
Comment (dutow) [inline docs/auto-enable.md:18]: This is just stylistic, but this example uses the `psql` command, while later examples only show SQL statements

File: docs/auto-enable.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   UPDATE pg_database SET datistemplate=true WHERE datname='my-template' 
+   ```
+
+4. Create a database using the new template
+
+    ```sql
+    CREATE DATABASE dbtest TEMPLATE my-template;
+    ```
+
+5. Check that `pg_stat_monitor` is enabled for the database
```
Comment (dutow) [inline docs/auto-enable.md:33]: is accessible from the database

File: docs/setup.md [hunk 3]
```diff
@@ -43,7 +43,13 @@ CREATE EXTENSION pg_stat_monitor;
 
 After the setup is complete, you can see the stats collected by `pg_stat_monitor`.
 
-By default, `pg_stat_monitor` is created for the `postgres` database. To access the statistics from other databases, you need to create the extension for every database.
+By default, `pg_stat_monitor` is created for the `postgres` database. To access the statistics from other databases, you need to create the extension for every database. 
+
+!!! note
+
+    When you create a new database `newdb`, `pg_stat_monitor` is not enabled for it yet the statistics metrics are captured. You can see the metrics for the `newdb` database either when you query it from the existing database `mydb` or after you explicitly create the `pg_stat_monitor` extension for the `newdb` database.
```
Comment (dutow) [inline docs/setup.md:50]: not accessible from it?

File: docs/auto-enable.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Automatically create `pg_stat_monitor` for every newly created database
```
[the remark this answers is not in the document]
Comment (nastena1606) [reply docs/auto-enable.md:1]: I follow the way we documented it earlier. In initial setup we have the step "Create the extension". Aren't we doing the same thing here - create pgsm for the template to have it enabled for every new db?

File: docs/auto-enable.md (hunk 1 above)
Comment (nastena1606) [reply docs/auto-enable.md:18]: What would you suggest: stick to psql or advise to connect to psql and use SQL statements going forward?

File: docs/auto-enable.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+In PostgreSQL, a database is created by copying the [system template database](https://www.postgresql.org/docs/current/manage-ag-templatedbs.html) `template1`. To automatically create the `pg_stat_monitor` extension for every newly created database, you can create a new system template database, create the extension for it and define this template when you create the database.  
+
+Your user must have the superuser privileges.
+
+Here's how to do it:
+
+1. Create a database to serve as the template.
```
[the remark this answers is not in the document]
Comment (nastena1606) [reply docs/auto-enable.md:9]: Which path is better? i thought going with the new template is better as it prevents from messing things up with the template1. If anything goes wrong, one can just drop the new template and go with the system default going forward. Moreover, may there be the cases when not all dbs should be subject to monitoring?

File: docs/auto-enable.md (hunk 1 above)
Comment (dutow) [reply docs/auto-enable.md]: Looking at our existing documentation, it usually uses SQL statements, and doesn't even mention the "connect to the database using psql..." part. So I would follow that style. Our documentation site also has some syntax highlighting for it, which is another plus. And similarly, it's a good idea to keep sql keywords upcase (for example WHERE, CREATE EXTENSION, and so on).

File: docs/auto-enable.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dutow) [reply docs/auto-enable.md:1]: My concern here is that from "create", I would assume that unless  I do this, it won't work with every database. But that's not the case: it works with all databases by default, it's just not accessible from databases where the extension isn't created.

File: docs/auto-enable.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (dutow) [reply docs/auto-enable.md]: I'm not sure how common is the use of custom templates with postgres, I think most people just modify template1. That's also easier to do, and doesn't require users to specify a custom template for database creation. Both are valid, and we can mention both - but the template1 part should also be there.

"Moreover, may there be the cases when not all dbs should be subject to monitoring?" - that my issue with the "Create" mentions. All DBs are subject to monitoring, even without these steps.

File: docs/auto-enable.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (nastena1606) [reply docs/auto-enable.md]: Amended

File: docs/auto-enable.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (nastena1606) [reply docs/auto-enable.md]: Amended

**Review by dutow — APPROVED** [review]

File: docs/auto-enable.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
+# Make `pg_stat_monitor` automatically accessible for every newly created database
+
+In PostgreSQL, a database is created by copying the [system template database](https://www.postgresql.org/docs/current/manage-ag-templatedbs.html) `template1`. To automatically create the `pg_stat_monitor` extension for every newly created database, you can do the following:
```
Comment (shoffman-percona) [inline docs/auto-enable.md:3]: I'm not sure I understand what you're trying to say in the first sentence... but here's what I _think_ you're going for:

```suggestion
In PostgreSQL, when a database is created the default settings of that database come from the [system template database](https://www.postgresql.org/docs/current/manage-ag-templatedbs.html) `template1`. To automatically create the `pg_stat_monitor` extension for every newly created database, you can do one of the following:
```

File: docs/auto-enable.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+* modify the `template1` template database and create the extension for it. Then all databases you create have the `pg_stat_monitor` view available.
```
Comment (shoffman-percona) [inline docs/auto-enable.md:5]: create or enable the extension for it?

ETA: In reading a bit further down, the command is "create extension" but I think we're actually enabling the extension in the template DB _for the next created database_ at least that is how it makes sense to me.

File: docs/auto-enable.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+* create a new system template database and create the extension for it. You need to define this template when you create a database to have the `pg_stat_monitor` view available for it.
+
+For either option, your user must have the superuser privileges. In the following procedures, the `postgres` user is used:
```
Comment (shoffman-percona) [inline docs/auto-enable.md:8]:
```suggestion
For either option, your user must have the superuser privileges; the instructions below assume your are working as the `postgres` user:
```

**Review by shoffman-percona — COMMENTED** [review]
Tried to make suggestions on how to clarify.  If you agree with my assertion that we're "enabling" vs "creating" the extension in the template DB there's a few places that need to be updated.  

But this is exactly what I think we need to publish for our users...and I would almost think this should be the default mode for installation... i.e. have this actually be in the Install/Setup page so users don't have to stumble on it separately.

**Review by shoffman-percona — APPROVED** [review]
LGTM!

**Review by capri1989 — APPROVED** [review]

**Pull request merged**